File: archive/maximum-number-of-pairs-in-array.js
const numberOfPairs = function(nums) {
  const mem = {}

  nums.forEach(n => {
    if (mem[n] === undefined) {
      mem[n] = 1
    } else {
      mem[n]++
    }
  })
  let ans = [0, 0]
  Object.keys(mem).forEach(k => {
    ans[1]+=mem[k]%2
    ans[0]+=Math.floor(mem[k]/2)
  })

  return ans
};

console.log(numberOfPairs([1,3,2,1,3,2,2]))

File: go-archive/maximum-number-of-pairs-in-array.go
func numberOfPairs(nums []int) []int {
    mem:=map[int]int{}

    for _, v := range nums {
        if _, isHas := mem[v]; isHas {
            mem[v]++
        } else {mem[v] = 1}
    }

    pairs:=0; unpaired:=0;

    for _, v:= range mem {
        if v % 2 == 0 {pairs+=v/2} else {
            unpaired++
            pairs+=(v-1)/2
        }
    }

    return []int{pairs, unpaired}
}